### PR TITLE
Update wrapper components and more #65

### PR DESCRIPTION
**Pull request by Boehrsi**
**Link to the given issue**
#65

**Describe what the problem was / what the new feature is**

- Updated Gradle (Android)
- Added `final` where applicable (Android)
- Removed `core strings` (Android and iOS)
- Simplified event delegation (Android)
- Removed unused files (Android)
- Updated main.dart in the example project to allow reruns (Flutter)

### CODE REVIEW

File: android/src/main/java/com/openxchange/deltachatcore/NativeInteractionManager.java [hunk 1]
```diff
@@ -329,107 +317,34 @@ public long handleEvent(final int event, long data1, long data2) {
...
+        if (eventChannelHandler != null) {
+            eventChannelHandler.handleEvent(eventId, data1Object, data2Object);
+        }
+    }
+
+    private Object getDataObject(int eventId, long data) {
+        return data1IsString(eventId) ? dataToString(data) : data;
+    }
+
+    private void delegateError(int eventId, long data1, String error) {
```
Comment (Boehrsi) [inline android/src/main/java/com/openxchange/deltachatcore/NativeInteractionManager.java:349]: Intentional logic change. We now delegate the event id, data1 and the data2 field as string to the Flutter part. Before we delegated "400" (error event id), event id and data2 as string.

File: ios/Classes/SwiftDeltaChatCorePlugin.swift [hunk 2]
```diff
@@ -172,12 +170,6 @@ public class SwiftDeltaChatCorePlugin: NSObject, FlutterPlugin {
         result(UIApplication.version)
     }
 
-    fileprivate func setCoreStrings(methodCall: FlutterMethodCall, result: FlutterResult) {
-        if let coreStrings: CoreStrings.CoreStringsDictionary = methodCall.arguments as? CoreStrings.CoreStringsDictionary {
-            CoreStrings.sharedInstance.strings = coreStrings
-        }
-        result(nil)
-    }
```
Comment (ghost) [inline ios/Classes/SwiftDeltaChatCorePlugin.swift:180]: So, shouldn't we get rid of the `CoreStrings` class, too?

**Review by ghost — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/openxchange/deltachatcore/NativeInteractionManager.java (hunk 1 above)
Comment (ghost) [reply android/src/main/java/com/openxchange/deltachatcore/NativeInteractionManager.java:349]: I guess I have to implement it the same way on iOS.

Comment (Boehrsi) [reply android/src/main/java/com/openxchange/deltachatcore/NativeInteractionManager.java:349]: Makes sense, we should align the native platforms.

File: ios/Classes/SwiftDeltaChatCorePlugin.swift (hunk 2 above)
Comment (Boehrsi) [reply ios/Classes/SwiftDeltaChatCorePlugin.swift:180]: Definitely yes. Didn't saw it while navigating through the iOS project.

**Comment by Boehrsi** [comment]
@phranck or @CihanOezkan could one of you remove the CoreStrings class and adjust the structure of the returned error events? Please just push it onto this branch.

**Comment by ghost** [comment]
> @phranck or @CihanOezkan could one of you remove the CoreStrings class and adjust the structure of the returned error events? Please just push it onto this branch.

@Boehrsi I will do it tomorrow.

**Review by florianhaar — APPROVED** [review]
Looks fine for me

**Comment by Boehrsi** [comment]
@phranck as soon as the Swift part is added please remove the `warning` label.

**Review by ghost — APPROVED** [review]

**Pull request merged**